### PR TITLE
Add Parameter classes, clean up code, add unit tests

### PR DESCRIPTION
**Pull request by wiechula**
- Parameter classes for the different hardware and software parts were added
- The code was cleaned up (Remove FairLinks)
- Unit tests were added and updated

### CODE REVIEW

**Comment by dberzano** [comment]
Please ignore the macOS error for now, it's a test we're running and we expect it to become green (stable) at some point.

File: Detectors/TPC/base/include/TPCBase/ParameterDetector.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    /// \return TPC length [cm]
+    float getTPClength() const { return mTPClength; }
+
+    /// Get the pad capacitance
+    /// \return Pad capacitance [pF]
+    float getPadCapacitance() const { return mPadCapacitance; }
+
+
+  private:
+    // use old c++03 due to root
```
Comment (sawenzel) [inline Detectors/TPC/base/include/TPCBase/ParameterDetector.h]: Don't understand this comment; ROOT can understand C++11.

File: Detectors/TPC/base/include/TPCBase/ParameterDetector.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    ParameterDetector(const ParameterDetector&) {}
+    void operator=(const ParameterDetector&) {}
```
Comment (matthiasrichter) [inline Detectors/TPC/base/include/TPCBase/ParameterDetector.h]: this is dangerous, it forbids the use outside, but inside the class copy ctor and assignment operator can be used and don't do what they should. Either use `= delete` or don't implement anything by replacing `{}` with a simple semicolon.

Comment (wiechula) [inline Detectors/TPC/base/include/TPCBase/ParameterDetector.h]: Indeed this is a bug. I suggest Andreas simple delete these lines from all Parameter classes. I guess the original intend was to not implement copy and assignment, so putting a semicolon at the end.
In former root version the `= delete` didn't work I think in the dictionary generation.

Comment (matthiasrichter) [inline Detectors/TPC/base/include/TPCBase/ParameterDetector.h]: Ah, this also explains the comment above. However, with the Root version we require this shouldn't be a problem.
Deleting the lines would then mean to allow the compiler to create the methods. Is this the intention?

Comment (wiechula) [inline Detectors/TPC/base/include/TPCBase/ParameterDetector.h]: Yes. I don't see a reason to not allow those operators for the Parameter classes.
Therefore, I proposed to simply delete the lines.

**Comment by wiechula** [comment]
Are there more comments to this PR or can it be merged?

**Comment by sawenzel** [comment]
@wiechula : Will take a look now.

**Pull request merged**